Blog: the pricing guide I never had

jamesefhawkins opened the pull request:
## Changes

* adds a new post that walks through what we've learned about pricing

Alternative title idea: 'Counterintuitive lessons about pricing' - @andyvan-ph what do you think?

Todo

- @andyvan-ph quick high level feedback please
- needs polishing ie internal links adding and a read through

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,104 @@
+
+So, how did we get pricing out of the way?
+
+**A free for life tier** means users don't _ever_ have to part with cash (or user data) to use our product. We do this in two ways - (i) we offer an open source product with a huge range of features (ii) we offer a little free usage on cloud for free, forever. This means we can get into production and demonstrate value to users who need to build a business case and can't just pop in their card.
```
```suggestion
**A free for life tier** means users don't _ever_ have to part with cash (or user data) to use our product. We do this in two ways - (i) we offer an open source product with a huge range of features; (ii) we offer a little free usage on cloud for free, forever. This means we can get into production and demonstrate value to users who need to build a business case and can't just pop in their card.
```

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,104 @@
+* An enterprise cloud offering
+* Some way of capturing market share for _extremely high volume_ but _low marginal value_ users
+
+## The results
+
+TODO: insert chart just showing shape of revenue growth
+
+Enough said.
+
+## Pricing as a product
+
+Think of your pricing like a product - you want users to become paying customers. What problems do users have from where they are today, to becoming paying customers tomorrow?
+
+Solve those.
```
I feel like we should end this piece on the 'The results' rather than having that extra point at the end. Any way we could have 'Pricing as a product' at the start? Maybe we put that intro and the piece is about how we used that approach to "nail pricing"

jamesefhawkins replied:
Added

andyvan-ph commented:
Couple of other thoughts:

- I think it's important to point out we've increased the free event limit to 1 million, just to advertise that point. This could be done in the headline

- Still need to have a think about the headline. I wonder if we specifically call out dev tool in the headline. Something along the lines the 'The devtool pricing guide I wish I had two years ago' or something along those lines.

jamesefhawkins commented:
> I think it's important to point out we've increased the free event limit to 1 million, just to advertise that point. This could be done in the headline

I think this _definitely_ needs announcing but should be its own (short) post. I've held off from doing this until @camerondeleone confirms that we have a good view of our pipeline for paid self hosted customers in Hubspot (so if we get a larger boost by promoting this, that we are more efficient with those who sign up)

jamesefhawkins commented:
> I think it's important to point out we've increased the free event limit to 1 million, just to advertise that point. This could be done in the headline

I think this _definitely_ needs announcing but should be its own (short) post. I've held off from doing this until @camerondeleone confirms that we have a good view of our pipeline for paid self hosted customers in Hubspot (so if we get a larger boost by promoting this, that we are more efficient with those who sign up)

jamesefhawkins commented:
@andyvan-ph 

I'm happy to merge this if you are

- added results image
- incorporated feedback around last paragraph
- I think a slightly pithy title like this is ok (A Story about Pivots landed on Hacker News)

andyvan-ph commented:
@jamesefhawkins will take a look today and probably merge later

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+Just because it's easy to get started, doesn't mean your product has to be cheap.
+
+Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just 1 extra engineer to self build in a large company, you theoretically shold be able to charge >$100K annually. If you've competition, it depends what they charge. 
```
```suggestion
Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just one extra engineer to self build in a large company, you theoretically shold be able to charge >$100K annually. If you've competition, it depends what they charge. 
```

andyvan-ph replied:
```suggestion
Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just one extra engineer to self build in a large company, you theoretically shold be able to charge >$100K annually. If you've competition, it depends what they charge. 
```

andyvan-ph replied:
```suggestion
Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just one extra engineer to self build in a large company, you theoretically shold be able to charge >$100K annually. If you've competition, it depends what they charge. 
```

andyvan-ph replied:
```suggestion
Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just one extra engineer to self build in a large company, you theoretically shold be able to charge >$100K annually. If you've competition, it depends what they charge. 
```

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+Early on, optimize for learning. Later on, optimize for the best solution for your users.
+
+**We didn't start with a $0 minimum**. Last summer, we worked on finding just five reference customers for our paid product (we now have X customers).
```
Is the (we now have X customers) a reference to something or is this a number that needs filling in?

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+## Make pricing work for your biggest customers
+
+We implemented logarithmically-decling pricing with usage. This means something like - every time your volume goes up 10x, the incremental price decrease 5x.
```
```suggestion
We implemented logarithmically-declining pricing with usage. This means something like - every time your volume goes up 10x, the incremental price decrease 5x.
```

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+If you like what you're reading, you'll probably have had a peek at [PostHog's pricing](../pricing).
+
+Before you copy paste, we've room for improvement. Here are a few things we're working on next.
```
```suggestion
Before you copy paste, we've room for improvement. Here are a few things we're working on next:
```

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+Before you copy paste, we've room for improvement. Here are a few things we're working on next.
+
+* Transparent enterprise pricing. We're currently figuring out how to standardize this by talking to users.
+* More clarity on about what you get if you upgrade.
```
```suggestion
* More clarity about what you get if you upgrade.
```

andyvan-ph commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+hideAnchor: true
+categories: ["CEO diaries", "Inside PostHog"]
+author: ["james-hawkins"]
+featuredImage: ../images/blog/product-people.png
```
```suggestion
featuredImage: ../images/blog/posthog-ceo-diary-blog.png
```

charlescook-ph commented:
> * Counterintuitive lessons about devtool pricing

This one (spelled correctly) would give the most value to our audience I think

pauldambra commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+So, how did we get pricing out of the way?
+
+**A free for life tier** means users don't _ever_ have to part with cash (or user data) to use our product. We do this in two ways - (i) we offer an open source product with a huge range of features; (ii) we offer a little free usage on cloud for free, forever. This means we can get into production and demonstrate value to users who need to build a business case and can't just pop in their card.
```
```suggestion
**A free for life tier** means users don't _ever_ have to part with cash (or user data) to use our product. We do this in two ways - (i) we offer an open source product with a huge range of features; (ii) we offer a little usage on cloud for free, forever. This means we can get into production and demonstrate value to users who need to build a business case and can't just pop in their card.
```

Don't say free twice so close together?

joethreepwood replied:
```suggestion
**A free-for-life tier** means users don't _ever_ have to part with cash (or user data) to use our product. We do this in two ways - (i) we offer an open source product with a huge range of features; (ii) we offer a little free usage on cloud for free, forever. This means we can get into production and demonstrate value to users who need to build a business case and can't just pop in their card.
```

pauldambra commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+Just because it's easy to get started, doesn't mean your product has to be cheap.
+
+Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just one extra engineer to self build in a large company, you theoretically shold be able to charge >$100K annually. If you've competition, it depends what they charge. 
```
```suggestion
Some companies with usage pricing get criticized for the escalating costs at scale. If it takes just one extra engineer to self build in a large company, you theoretically should be able to charge >$100K annually. If you've competition, it depends what they charge. 
```

pauldambra commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+## Make pricing work for your biggest customers
+
+We implemented logarithmically-declining pricing with usage. This means something like - every time your volume goes up 10x, the incremental price decrease 5x.
```
This section feels short... is it part of the section before? i.e. we don't price to value we do this instead.

andyvan-ph replied:
I've adding a little more context so it doesn't feel too abrupt

pauldambra reviewed (approved):
Couple of typos and a question, but other than that 👍

joethreepwood commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+featuredImageType: full
+---
+
+Imagine being able to grow twice as fast with a few hours of work. Changing the way you price has a real chance to get you there. 
```
```suggestion
Imagine being able to grow twice as fast with just a few hours of work. Changing your pricing has a real chance to get you there. 
```

joethreepwood commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+## Don't let pricing get in the way of adoption
+
+Most software companies agonize over how their product is adopted yet when it comes to pricing, they default to charging a certain recurring amount per month. The only thing under deep consideration is how _much_... not _how_.
```
```suggestion
Most software companies agonize over how their product is adopted yet when it comes to pricing, they default to charging a certain recurring amount per month. They only thing consider how _much_ to charge, not _how_ to charge.
```

Passive voice and weird syntax imho

joethreepwood commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+Most software companies agonize over how their product is adopted yet when it comes to pricing, they default to charging a certain recurring amount per month. The only thing under deep consideration is how _much_... not _how_.
+
+Our revelation was realizing that, for us, pricing shouldn't get in the way of adoption, because we don't sell top down into companies.. our most successful paying customers get there as follows:
```
```suggestion
We realized that our pricing shouldn't get in the way of adoption, because we don't sell top-down into companies. Instead, our most successful paying customers get there as follows:
```
I think this is just simpler construction

joethreepwood commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+**A free for life tier** means users don't _ever_ have to part with cash (or user data) to use our product. We do this in two ways - (i) we offer an open source product with a huge range of features; (ii) we offer a little free usage on cloud for free, forever. This means we can get into production and demonstrate value to users who need to build a business case and can't just pop in their card.
+
+**Encourage users to upgrade**, since you need people to do exactly that! We encourage people to put in their credit card - by offering our paid features starting at $0 (with a generous amount of free usage) if they do so. This lets users evaluate the paid product very easily, and it lets them upgrade with no friction if they're heavy users. This exchange feels fair to us - it helps us figure out ...
```
```suggestion
**Encourage users to upgrade**, since you need people to do exactly that! We encourage people to put in their credit card by offering our paid features starting at $0 (with a generous amount of free usage) if they do so. This lets users evaluate the paid product very easily, and upgrade with no friction if they're heavy users. This exchange feels fair to us - it helps us figure out who is serious versus playing around. Our numbers reflect that users feel the same way - over half our users enter their card details.
```

This doesn't need separating, it can be one sentence.

joethreepwood commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+**Encourage users to upgrade**, since you need people to do exactly that! We encourage people to put in their credit card - by offering our paid features starting at $0 (with a generous amount of free usage) if they do so. This lets users evaluate the paid product very easily, and it lets them upgrade with no friction if they're heavy users. This exchange feels fair to us - it helps us figure out ...
+
+**No minimum spend**, so paying users can start small if they want to. This means a customer can gradually get more committed to you. If customers ask for trials, you can get them straight into a paying relationship "just keep the volume low", with a seamless way to upgrade - use the product more. That makes it easier for customers to avoid some complex processes around pilots/proofs of concept,  ...
```
```suggestion
**No minimum spend**, so paying users can start small if they want to. This means a customer can gradually get more committed to you. If customers ask for trials, you can get them straight into a paying relationship ("just keep the volume low"), with a seamless way to upgrade - use the product more. That makes it easier for customers to avoid some complex processes around pilots/proofs of concept, and it stops a jarring discussion around why someone is going from $0/month to $10k/month. Don't let your pricing generate a stressful decision point.
```
The brackets make this construction MUCH easier to parse for me.

joethreepwood commented on contents/blog/pricing-lessons.md:
```diff
@@ -0,0 +1,96 @@
+
+**No minimum spend**, so paying users can start small if they want to. This means a customer can gradually get more committed to you. If customers ask for trials, you can get them straight into a paying relationship "just keep the volume low", with a seamless way to upgrade - use the product more. That makes it easier for customers to avoid some complex processes around pilots/proofs of concept,  ...
+
+**No annual contracts**, unless a customer requests one. This cuts the risk down for customers, reducing the need for intensive evaluation periods. Ironically, the less commitment you ask for, the more you may get. Of course, some customers _will_ want you to do this, to lock in pricing, but don't default to this.
```
```suggestion
**No annual contracts**, unless a customer requests one. This cuts the risk down for customers and reduces the need for intensive evaluation periods. Ironically, the less commitment you ask for, the more you may get. Of course, some customers _will_ want you to do this, to lock in pricing, but don't default to this.
```
Tense correction.

joethreepwood reviewed (commented):
Skimmed through and made a few quick suggestions/corrections for you to take or leave. 

Title-wise, I'm a fan of keeping it simple, both in construction and in communicating the meaning. _Counterintuitive lessons about devtool pricing_ is probably fine, but _Five surprising lessons about SaaS pricing_ and _Five SaaS pricing lessons we learned the hard way_ feels a bit more specific and click-worthy in that vein. 

The title I'd actually _want_ to propose would be much simpler, something like: 'Why we give PostHog away for free' -- but I think that may be a bit too close to [this current HN-frontpager.](https://tailscale.com/blog/free-plan/)

andyvan-ph commented:
> Skimmed through and made a few quick suggestions/corrections for you to take or leave.
> 
> Title-wise, I'm a fan of keeping it simple, both in construction and in communicating the meaning. _Counterintuitive lessons about devtool pricing_ is probably fine, but _Five surprising lessons about SaaS pricing_ and _Five SaaS pricing lessons we learned the hard way_ feels a bit more specific and click-worthy in that vein.
> 
> The title I'd actually _want_ to propose would be much simpler, something like: 'Why we give PostHog away for free' -- but I think that may be a bit too close to [this current HN-frontpager.](https://tailscale.com/blog/free-plan/)

I quite like 'Why we give PostHog away for free', too.

Think it's between that and 'Counterintuitive lessons about devtool pricing'.

charlescook-ph commented:
Counter-argument: 'giving away for free' doesn't feel like a natural way to describe an open source/core product. I'd expect that term to refer to a regular piece of software that you've decided to do a free version of. 

'Why did we give it away for free?' 'Er because that's how open source works by definition'.

andyvan-ph commented:
> Counter-argument: 'giving away for free' doesn't feel like a natural way to describe an open source/core product. I'd expect that term to refer to a regular piece of software that you've decided to do a free version of.
> 
> 'Why did we give it away for free?' 'Er because that's how open source works by definition'.

Fair point. 

Getting down to a final two, I'd say either:

1. Counterintuitive lessons about devtool pricing
2. Why you shouldn't 'price to value' and other counterintuitive pricing lessons

Throwing '2' into the mix as, at least for HN, I feel like it's good to have something in the title someone might diasgree/agree with to create debate.

jamesefhawkins commented:
> Why you shouldn't 'price to value' and other counterintuitive pricing lessons

Not pricing to value isn't the _main_ thrust of this article (giving stuff away for free is, but I agree with the objections on that title above).

> Counterintuitive lessons about devtool pricing

I think this is the best one thus far.

Final counter - I think 'Counterintuitive lessons about our pricing' - makes it clear this is speaking from experience and that we don't think this is applicable to any devtool (it isn't), and it highlights transparency so is what I'd like to go with

andyvan-ph commented:
Yup, works for me.

andyvan-ph commented:
Merged
